### PR TITLE
Page: Add Job Lead

### LINKED ISSUE #60: Job Lead add page
Implement job lead page based on the figma. Should be connected to backend.

You will also require implementing some more functionality related to getting employers. Just making explicit that is within the scope of this ticket.

Talk to designers throughout ticket for clarification, and also check with them before the PR is opened to verify everything is matching.

[figma link](https://www.figma.com/file/ckkCZEVkae0kIXUwaQPbib/High-Fidelity-Designs?node-id=6%3A28&t=0zBeBMgl6kj9tzlX-0)

### Notes
* design for Dropdown for employer may be altered during this ticket, but you should be kept in loop if u talk to designers.

### PR DESCRIPTION
**Pull request by helen-li**
# Description

Fixes #60 
* Frontend change: added components for the Add Job Lead page (screenshot below)
* Backend change: added Employer GET endpoint to retrieve all docs in the employer collection, also fixed errors caused by #58 due to the collection not being connected in `conn.ts` and `index.ts` in the server dir
* Dependencies: `@mui/x-date-pickers` for date picker component, `date-fns` for handling date objects easier

## Sample curl
curl --location --request GET 'http://localhost:3001/api/employer/'

## Response
[
    {
        "_id": "63d1d945e1130364473e7bfa",
        "name": "Test ",
        "public_name": "Test",
        "contacts": [],
        "creator_id": "63bb1f16939f28affbacf5fa",
        "owner_id": "63bb1f16939f28affbacf5fa",
        "legal_name": null,
        "hr_contacts": null,
        "secondary_address_id": null,
        "initial_contact": null,
        "id": null,
        "timeline": null
    },
    {
        "_id": "63d1d947e1130364473e7bfb",
        "name": "Test 1",
        "public_name": "Test",
        "contacts": [],
        "creator_id": "63bb1f16939f28affbacf5fa",
        "owner_id": "63bb1f16939f28affbacf5fa",
        "legal_name": null,
        "hr_contacts": null,
        "secondary_address_id": null,
        "initial_contact": null,
        "id": null,
        "timeline": null
    }
]

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [ ] I have run `npm run lint:fix`
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (if required)
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works (Delete if frontend)

### CODE REVIEW

**Review by annie951977 — CHANGES_REQUESTED** [review]
The way you format your code makes it very readable, so that's awesome! 

I tried compiling your branch and aside from package lock stuff, I also got these errors. Let me know if you run into these problems as well or if there's something that's weird on my end

image
image
image

File: client/src/App.tsx [hunk 1]
```diff
@@ -16,6 +18,8 @@ function App() {
 		<div className="App">
 			<header className="App-header">
 				{Communication()}
+				{ActivityTimeline()}
+				{AddJobLead()}
```
[the remark this answers is not in the document]
Comment (helen-li) [reply client/src/App.tsx:22]: Just for testing! I'm not entirely sure whether router is set up for client yet to navigate between pages so I just threw it in here, can totally remove before I merge

**Comment by helen-li** [comment]
> The way you format your code makes it very readable, so that's awesome!
> 
> I tried compiling your branch and aside from package lock stuff, I also got these errors. Let me know if you run into these problems as well or if there's something that's weird on my end
> 
> image image image

@Manal-jpg Could you try compiling on your end and see if you can reproduce these errors? I somehow could not reproduce the same warnings on my system and am concerned there's something wrong with my setup. Thanks!

For reference, my output just looks like this:

**Review by Manal-jpg — APPROVED** [review]
I've finally reviewed the PR, sorry for the long wait. Everything looks great to me. I didn't get the errors Annie was having, and it was running on my computer without a problem (there was a small package error that Ganesh addressed on Slack). As for the employer conn.ts problem, I just realized that I accidentally removed my changes for that file while solving merge conflicts so sorry about that, and tysm for fixing it.

**Comment by edwardhuahan** [comment]
I just compiled everything, it gave me a few warnings but no errors

image

**Comment by edwardhuahan** [comment]
Awesome job! It looks great, I think it'd be best to style your components using css files like we decided instead of the sx prop but just to save time I think it's fine this time

**Review by edwardhuahan — APPROVED** [review]
Make sure to make external css files next time for styling

**Review by annie951977 — APPROVED** [review]
Other that the previously addressed issues which look like are not replicable, looks good to me!

**Pull request merged**